### PR TITLE
feat: added contextprovider for theme and useTheme hook

### PR DESCRIPTION
**Pull request by christianblandford**

### CODE REVIEW

**Comment by christianblandford** [comment]
@benjitrosch can you give me a quick sanity check on this before we merge

File: src/Theme/Theme.stories.tsx [hunk 1]
```diff
@@ -12,32 +14,21 @@ export default {
...
-
-      <Theme dataTheme="lofi" className="p-8 bg-neutral rounded-2xl">
-        <div className="flex gap-x-2">
-          <Button>Default</Button>
-          <Button color="primary">Primary</Button>
-          <Button color="secondary">Secondary</Button>
-          <Button color="accent">Accent</Button>
-        </div>
-      </Theme>
+      {DEFAULT_THEMES.map((theme) => (
```
Comment (benjitrosch) [inline src/Theme/Theme.stories.tsx:17]: Nice 😄

**Review by benjitrosch — APPROVED** [review]
LGTM!

File: src/Theme/Theme.stories.tsx (hunk 1 above)
Comment (christianblandford) [reply src/Theme/Theme.stories.tsx:17]: TY, going to remove the test story and then merge

**Pull request merged**